src/wccp/WccpConfig.cc

```diff
@@ -712,86 +630,96 @@ CacheImpl::loadServicesFromFile(const char *path)
     }
 
     // Get any group specific routers.
-    routers.clear(); // reset list.
-    if ((prop = svc_cfg[SVC_PROP_ROUTERS]).hasValue()) {
-      ts::Errata status = load_routers(prop, routers);
+    std::vector<uint32_t> routers;
+    YAML::Node routers_prop = svc_cfg[SVC_PROP_ROUTERS];
+    if (routers_prop) {
+      ts::Errata status = load_routers(routers_prop, routers);
       if (!status)
         zret.push(ts::Errata::Message(23, LVL_INFO, "Router specification invalid.").set(status));
     }
-    if (!routers.size() && !Seed_Router.size())
-      zret.push(No_Valid_Routers(svc_line));
+
+    if (!routers.size() && !Seed_Router.size()) {
+      zret.push(No_Valid_Routers(routers_prop.Mark().line));
+    }
 
     // See if can proceed with service group creation.
     ts::Errata::Code code = code_max(zret);
     if (code >= LVL_WARN) {
-      zret = Unable_To_Create_Service_Group(svc_line).set(zret);
+      zret = Unable_To_Create_Service_Group(svc_cfg.Mark().line).set(zret);
       return zret;
     }
 
     // Properties after this are optional so we can proceed if they fail.
     GroupData &svc = this->defineServiceGroup(svc_info);
 
     // Is there a process we should track?
-    if ((prop = svc_cfg[SVC_PROP_PROC]).hasValue()) {
-      if (ts::config::StringValue == prop.getType()) {
-        svc.setProcName(prop.getText());
-      } else {
-        zret.push(Prop_Invalid_Type(prop, ts::config::StringValue));
-      }
+    YAML::Node proc_name = svc_cfg[SVC_PROP_PROC];
+    if (proc_name) {
+      std::string text = proc_name.as<std::string>();
+      svc.setProcName(ts::ConstBuffer(text.c_str(), text.size()));
```

Comment:
As long as you're cleaning up, get rid of `ConstBuffer` entirely. Use `string_view`. If you change `svc.setProcName` to take a `string_view` then you can simply pass `text` from here.